invenio_rdm_records/services/components/access.py

```diff
@@ -19,13 +20,31 @@ class AccessComponent(ServiceComponent):
 
     def _populate_access_and_validate(self, identity, data, record, **kwargs):
         """Populate and validate the record's access field."""
+        errors = []
         if record is not None and "access" in data:
+            access = data.get("access")
```

Comment:
ultra minor:
```suggestion
            access = data.get("access", {})
```

To avoid the `(access or {})` below.